clusterous/cluster.py

```diff
@@ -685,49 +840,78 @@ def init_cluster(self, cluster_name, cluster_spec, nodes_info=[], logging_level=
                     self._logger.error('Unable to create S3 bucket due to an AWS conflict. Try again later.')
                     self._logger.error(e)
                     raise ClusterException('Unrecoverable error while trying to start cluster')
-
-            # Shared volume
-            if shared_volume_id:
-                try:
-                    shared_volume = conn.get_all_volumes([shared_volume_id])[0]
-                    if shared_volume.status != 'available':
-                        raise ClusterException('Volume "{0}" is not available'.format(shared_volume_id))
-                    vpc_conn = boto.vpc.connect_to_region(c['region'], aws_access_key_id=c['access_key_id'],aws_secret_access_key=c['secret_access_key'])
-                    subnet = vpc_conn.get_all_subnets([c['subnet_id']])[0]
-                    if subnet.availability_zone != shared_volume.zone:
-                        raise ClusterException('Conflict in availability zone. Subnet "{0}" in "{1}" and Volume "{2}" in "{3}"'.format(c['subnet_id'],
-                                                                                                                                subnet.availability_zone,
-                                                                                                                                shared_volume_id,
-                                                                                                                                shared_volume.zone))
-                except boto.exception.EC2ResponseError as e:
-                    raise ClusterException('Volume "{0}" does not exist'.format(shared_volume_id))
-
-            # Create Security group
-            self._logger.info('Creating security group')
-            sg_id = self._create_security_group(cluster_name, conn)
-
+    
             # Cluster info file: Having created the first cluster resource, set cluster name and flag
             self._set_cluster_info({'cluster_name': cluster_name, 'running': False})
-
-            #
-            # Launch all instances
-            #
-
+    
+            # Configuring VPC
+            self._logger.info('Configuring VPC')
+            self.vpc_conn = vpc_conn
```

Comment:
This is a member variable that is only ever valid if init_cluster is called first. Since vpc_conn is only ever used by methods called by init_cluster, then it should be passed as a method parameter. It has no validity outside the working of the init_cluster method.